### ISSUE TITLE
bug(s?) in combine

### ISSUE THREAD

**Issue by abuts** [pace-neutrons/Horace#1778]
Info provided by Alex in connection with the user's description below:

data in /mnt/ceph/instrument/MERLIN/RBNumber/RB2220429/sqw 
(https://pace-neutrons.slack.com/archives/C05E5KBQE9X/p1733156850078569)
cut3 (two with offset) in the letter are certainly wrong
data just cut, not folded as it should be 

User requests as follows:

Dear horace-team,

I wanted to combine some sqw objects to enhance statistics, combine features and for smooth plots of "Bloch equivalent" directions. However, the results do not look as expected.

I have three individual example cuts for three different Gamma-R directions ([1,1,1] in simlpe cubic) all starting from the [2,2,2] Bragg peak used as uoffset:
![8f0SOaXLsYQezdQy](https://github.com/user-attachments/assets/92d1a488-54e5-453f-a787-13f3f1e9863a)
Next figures show the combination of
left: 1+2+3
center: 1+2
right: 2+3
![3q9dXUdiuGZf9Yt6](https://github.com/user-attachments/assets/469ac68f-7f41-46dc-9c76-9d3738aa3a8e)
There appear extra features at ~0.18 and -0.2 that I do not have in the individual cuts. I expected a nice overlay of the figures 1+2+3.

This is my matlab code is:
```Matlab
sqw_file=['H:/' 'STO_300K_35meV_align.sqw'];
r1 = cut_sqw(sqw_file, struct('u', [1.0, 1.0, 1.0],  'v', [1.0, 1.0, -2.0],  'type', 'rrr', 'uoffset', [2.0, 2.0, 2.0, 0.0]), [0.0, 0.025, 0.5], [-0.05, 0.05], [-0.05, 0.05], [-10.0, 0.5, 30.0]);
r2 = cut_sqw(sqw_file, struct('u', [-1.0, 1.0, 1.0], 'v', [-1.0, 1.0, -2.0], 'type', 'rrr', 'uoffset', [2.0, 2.0, 2.0, 0.0]), [0.0, 0.025, 0.5], [-0.05, 0.05], [-0.05, 0.05], [-10.0, 0.5, 30.0]);
r3 = cut_sqw(sqw_file, struct('u', [1.0, -1.0, 1.0], 'v', [1.0, -1.0, -2.0], 'type', 'rrr', 'uoffset', [2.0, 2.0, 2.0, 0.0]), [0.0, 0.025, 0.5], [-0.05, 0.05], [-0.05, 0.05], [-10.0, 0.5, 30.0]);
%figure
r1_c = compact(r1)
r2_c = compact(r2)
r3_c = compact(r3)
close all
plot(r1_c)
caxis([0, 0.02]);
keep_figure
%figure
plot(r2_c)
caxis([0, 0.02]);
keep_figure
plot(r3_c)
caxis([0, 0.02]);
keep_figure
result123 = combine_sqw(r1_c, r2_c, r3_c)
plot(result123)
caxis([0, 0.02]);
keep_figure
result12 = combine_sqw(r1_c, r2_c)
plot(result12)
caxis([0, 0.02]);
keep_figure
result23 = combine_sqw(r2_c, r3_c)
plot(result23)
caxis([0, 0.02]);
keep_figure
```
If I use a horace 4.0 file as input of the above cuts I get the following error message:

>> combine_sqw_test
*** Step 1 of 2; Read data for  5003707 pixels -- processing data... ----->  retained    677073 pixels
*** Step 2 of 2; Read data for  1937482 pixels -- processing data... ----->  retained    183464 pixels
*** Step 1 of 2; Read data for  5009343 pixels -- processing data... ----->  retained    667142 pixels
*** Step 2 of 2; Read data for  3895476 pixels -- processing data... ----->  retained    560247 pixels
*** Step 1 of 2; Read data for  5006109 pixels -- processing data... ----->  retained    600319 pixels
*** Step 2 of 2; Read data for  4731868 pixels -- processing data... ----->  retained    568178 pixels
 

  2-dimensional object:
 -------------------------
 Original datafile:  <none>
             Title: <none>

 Lattice parameters (Angstroms and degrees):
     a=3.898          b=3.898           c=3.898      
 alpha=90          beta=90          gamma=90         

 Extent of data:
 Number of spe files: 77
    Number of pixels: 860537

 Size of 2-dimensional dataset: [21x64]
     Plot axes:
          \zeta = -0.0125:0.025:0.5125 in [2+\zeta, 2+\zeta, 2+\zeta]
          E = -3.75:0.5:28.25
     Integration axes:
          -0.05 =< \xi =< 0.05 in [2+0.5\xi, 2+0.5\xi, 2-\xi]
          -0.05 =< \eta =< 0.05 in [2-\eta, 2+\eta, 2]
 Object is: memory based

 

  2-dimensional object:
 -------------------------
 Original datafile:  <none>
             Title: <none>

 Lattice parameters (Angstroms and degrees):
     a=3.898          b=3.898           c=3.898      
 alpha=90          beta=90          gamma=90         

 Extent of data:
 Number of spe files: 85
    Number of pixels: 1227389

 Size of 2-dimensional dataset: [21x64]
     Plot axes:
          \zeta = -0.0125:0.025:0.5125 in [2-\zeta, 2+\zeta, 2+\zeta]
          E = -3.75:0.5:28.25
     Integration axes:
          -0.05 =< \xi =< 0.05 in [2-0.5\xi, 2+0.5\xi, 2-\xi]
          -0.05 =< \eta =< 0.05 in [2-\eta, 2-\eta, 2]
 Object is: memory based

 

  2-dimensional object:
 -------------------------
 Original datafile:  <none>
             Title: <none>

 Lattice parameters (Angstroms and degrees):
     a=3.898          b=3.898           c=3.898      
 alpha=90          beta=90          gamma=90         

 Extent of data:
 Number of spe files: 85
    Number of pixels: 1168497

 Size of 2-dimensional dataset: [21x64]
     Plot axes:
          \zeta = -0.0125:0.025:0.5125 in [2+\zeta, 2-\zeta, 2+\zeta]
          E = -3.75:0.5:28.25
     Integration axes:
          -0.05 =< \xi =< 0.05 in [2+0.5\xi, 2-0.5\xi, 2-\xi]
          -0.05 =< \eta =< 0.05 in [2+\eta, 2+\eta, 2]
 Object is: memory based

Error using check_combo_arg_
Number of instruments: 169 is not equal to number of runs: 170;
 Number of samples 169 is not equal to number of runs: 170;
 Number of detector arrays 169 is not equal to number of runs: 170;

Error in Experiment/check_combo_arg (line 661)
            obj = check_combo_arg_(obj);

Error in set_positional_and_key_val_arguments_ (line 188)
obj=obj.check_combo_arg();

Error in serializable/set_positional_and_key_val_arguments (line 110)
    set_positional_and_key_val_arguments_(obj,...

Error in init_ (line 40)
    [obj,remains] = set_positional_and_key_val_arguments(obj,...

Error in Experiment (line 117)
            obj = init_(obj,varargin{:});

Error in combine_experiments_ (line 95)
exp = Experiment(det, instr, sampl,expinfo);

Error in Experiment.combine_experiments (line 623)
            [exp,nspe] = combine_experiments_(exp_cellarray,allow_equal_headers,keep_runid);

Error in sqw/combine_sqw (line 141)
wout.experiment_info = exper{1}.combine_experiments(exper(2:end),true,true);

Error in combine_sqw_test (line 23)
result123 = combine_sqw(r1_c, r2_c, r3_c)

**Comment by abuts** [comment]
fixed by #1783

**Comment by abuts** [comment]
fixed by Re #1783
